domain/machine/service/service.go

```diff
@@ -394,6 +404,24 @@ func (s *Service) GetMachineUUID(ctx context.Context, name coremachine.Name) (st
 	return s.st.GetMachineUUID(ctx, name)
 }
 
+// LXDProfiles returns the names of the LXD profiles on the machine.
+func (s *Service) LXDProfiles(ctx context.Context, mUUID string) ([]string, error) {
```

Comment:
This was always a source of ambiguity in the old state.

I think these method names should be:
- `AppliedLXDProfileNames`
- `SetAppliedLXDProfileNames`